pkg/kubelet/kubelet.go

```diff
@@ -699,11 +699,16 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		}
 		// TODO(roberthbailey): Can we do this without having credentials to talk
 		// to the cloud provider?
-		instanceID, err := instances.ExternalID(kl.hostname)
+		externalID, err := instances.ExternalID(kl.hostname)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get external ID from cloud provider: %v", err)
+		}
+		node.Spec.ExternalID = externalID
+		instanceID, err := instances.InstanceID(node.Name)
```

Comment:
@bgrant0607 this looks like a problem in the current (re-)implementation. As you mentioned we can't expect all the nodes to be able to talk with the cloud provider. Any suggestion?